server/src/main/java/io/seata/server/storage/redis/lock/RedisLocker.java

```diff
@@ -0,0 +1,222 @@
+import redis.clients.jedis.Jedis;
+
+/**
+ * @author funkye
+ */
+public class RedisLocker extends AbstractLocker {
+
+    private static final Integer DEFAULT_SECONDS = 30;
+
+    private static final String DEFAULT_REDIS_SEATA_LOCK_PREFIX = "SEATA_LOCK_";
+
+    private static final String DEFAULT_REDIS_SEATA_LOCK_XID_PREFIX = "SEATA_LOCK_XID_";
+
+    /**
+     * Instantiates a new Redis locker.
+     *
+     */
+    public RedisLocker() {}
+
+    @Override
+    public boolean acquireLock(List<RowLock> rowLocks) {
+        if (CollectionUtils.isEmpty(rowLocks)) {
+            // no lock
+            return true;
+        }
+        List<String> successList = new ArrayList<>();
+        long status = 0;
+        try (Jedis jedis = JedisPooledFactory.getJedisInstance()) {
+            List<LockDO> locks = convertToLockDO(rowLocks);
+            for (LockDO lock : locks) {
+                String key = DEFAULT_REDIS_SEATA_LOCK_PREFIX + lock.getRowKey();
+                status = jedis.setnx(key, JSON.toJSONString(lock));
+                if (status == 1) {
+                    successList.add(key);
+                    jedis.lpush(DEFAULT_REDIS_SEATA_LOCK_XID_PREFIX + lock.getXid(), key);
+                    jedis.expire(key, DEFAULT_SECONDS);
+                } else {
+                    break;
+                }
+            }
+            if (status != 1) {
+                String[] rms = successList.toArray(new String[successList.size()]);
+                if (rms.length > 0) {
+                    jedis.del(rms);
+                }
+                return false;
+            } else {
+                return true;
+            }
+        }
+    }
+
+    @Override
+    public boolean releaseLock(List<RowLock> rowLocks) {
+        if (CollectionUtils.isEmpty(rowLocks)) {
+            // no lock
+            return true;
+        }
+        String[] keys = new String[rowLocks.size()];
+        List<LockDO> locks = convertToLockDO(rowLocks);
+        for (int i = 0; i < locks.size(); i++) {
+            String key = DEFAULT_REDIS_SEATA_LOCK_PREFIX + locks.get(i).getRowKey();
+            keys[i] = key;
+        }
+        try (Jedis jedis = JedisPooledFactory.getJedisInstance()) {
+            jedis.del(keys);
+            return true;
+        }
+    }
+
+    @Override
+    public boolean releaseLock(String xid, List<Long> branchIds) {
+        if (CollectionUtils.isEmpty(branchIds)) {
+            // no lock
+            return true;
+        }
+        try (Jedis jedis = JedisPooledFactory.getJedisInstance()) {
+            String lockListKey = DEFAULT_REDIS_SEATA_LOCK_XID_PREFIX + xid;
+            Set<String> keys = new HashSet<>();
+            List<String> redisLockJson = null;
+            int start = 0;
+            do {
+                redisLockJson = jedis.lrange(lockListKey, start, 100);
+                if (null != redisLockJson) {
+                    keys.addAll(redisLockJson);
+                    start = keys.size();
```

Comment:
the list size is 201, the `start` not change?

Reply:
ok